adaptive/learner/balancing_learner.py

```diff
@@ -111,38 +111,57 @@ def strategy(self, strategy):
             raise ValueError(
                 'Only strategy="loss_improvements", strategy="loss", or'
                 ' strategy="npoints" is implemented.')
+        self._points = {}  # reset the cache
 
     def _ask_and_tell_based_on_loss_improvements(self, n):
-        points = []
-        loss_improvements = []
+        chosen_points = []
+        chosen_loss_improvements = []
+        npoints_per_learner = defaultdict(int)
```

Comment:
Rename to `n_selected` or something like that

Add a comment along the lines of "`same as Counter(i[0] for i in selected)`"

Reply:
The code is no longer the same, so I resolved it.